src/math/lp/lar_solver.cpp

```diff
@@ -1466,64 +1387,41 @@ namespace lp {
 
     // below is the initialization functionality of lar_solver
 
-    bool lar_solver::strategy_is_undecided() const {
-        return m_settings.simplex_strategy() == simplex_strategy_enum::undecided;
-    }
-
-    var_index lar_solver::add_named_var(unsigned ext_j, bool is_int, const std::string& name) {
-        var_index j = add_var(ext_j, is_int);
+    lpvar lar_solver::add_named_var(unsigned ext_j, bool is_int, const std::string& name) {
+        lpvar j = add_var(ext_j, is_int);
         m_var_register.set_name(j, name);
         return j;
     }
 
-    unsigned lar_solver::external_to_column_index(unsigned ext_j) const {
-        unsigned j = external_to_local(ext_j);
-        if (j == null_lpvar)
-            return j;
-
-        if (tv::is_term(j))
-            return map_term_index_to_column_index(j);
-
-        return j;
-    }
-
     struct lar_solver::undo_add_column : public trail {
         lar_solver& s;
         undo_add_column(lar_solver& s) : s(s) {}
         void undo() override {
+            auto& col = s.m_columns.back();
+            if (col.term() != nullptr) {
+                if (s.m_need_register_terms)
+                    s.deregister_normalized_term(*col.term());
+                delete col.term();
```

Comment:
or delete s.m_terms.back();

Reply:
we have col == s.m_terms.back() here already. I am not sure what we gain by using "delete s.m_terms.back()".